cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -2317,6 +2365,14 @@ const FunctionDoc utf8_lower_doc(
     "Transform input to lowercase",
     ("For each string in `strings`, return a lowercase version."), {"strings"});
 
+const FunctionDoc ascii_reverse_doc(
+    "Reverse ascii input",
+    ("For each ascii string in `strings`, return a reversed version."), {"strings"});
+
+const FunctionDoc utf8_reverse_doc(
+    "Reverse utf8 input",
+    ("For each utf8 string in `strings`, return a reversed version."), {"strings"});
```

Comment:
Should we clarify that the reversing is done at the codepoint level, and not the grapheme cluster level? (So people would be aware that this kernel can't handle an é encoded as e + ´, for instance, or a flag emoji.)